src/drivers/intel/ssp/ssp.c

```diff
@@ -711,6 +719,54 @@ static int ssp_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	ssp->state[DAI_DIR_PLAYBACK] = COMP_STATE_PREPARE;
 	ssp->state[DAI_DIR_CAPTURE] = COMP_STATE_PREPARE;
 
+	if (config->flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) {
```

Comment:
the ssp_set_config() function is a mile long already. Do you think it will benefit if we moved the if/else for HW_PARAMS/HW_FREE to separate functions?

Reply:
we're already using helpers. Moving the code flow to a separate function isn't going to change much in terms of code size, maybe it'll be more readable. I'll follow the majority here.

Reply:
no change

Reply:
@plbossart With https://github.com/thesofproject/linux/pull/2965/commits/e7fdfb2a1cf5e093890f405573a07b3ede0d9a15

```
/* DAI_CONFIG flags */
#define SOF_DAI_CONFIG_FLAGS_MASK	0x3
#define SOF_DAI_CONFIG_FLAGS_NONE	(0 << 0) /**< DAI_CONFIG sent without stage information */
#define SOF_DAI_CONFIG_FLAGS_HW_PARAMS	(1 << 0) /**< DAI_CONFIG sent during hw_params stage */
#define SOF_DAI_CONFIG_FLAGS_HW_FREE	(2 << 0) /**< DAI_CONFIG sent during hw_free stage */
#define SOF_DAI_CONFIG_FLAGS_RFU	(3 << 0) /**< not used, reserved for future use */
```
We are not supposed to use `config->flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS` I would suggest to use 
```
switch (config->flags & SOF_DAI_CONFIG_FLAGS_MASK)
        case SOF_DAI_CONFIG_FLAGS_HW_PARAMS:
...
```